packages/stream_video_noise_cancellation/android/src/main/kotlin/io/getstream/video/flutter/stream_video_noise_cancellation/StreamVideoNoiseCancellationPlugin.kt

```diff
@@ -0,0 +1,129 @@
+package io.getstream.video.flutter.stream_video_noise_cancellation
+
+import android.content.Context
+import android.content.pm.PackageManager
+import android.util.Log
+import io.flutter.embedding.engine.plugins.FlutterPlugin
+import io.flutter.plugin.common.MethodCall
+import io.flutter.plugin.common.MethodChannel
+import io.flutter.plugin.common.MethodChannel.MethodCallHandler
+import io.flutter.plugin.common.MethodChannel.Result
+import io.getstream.webrtc.flutter.FlutterWebRTCPlugin
+import io.getstream.video.flutter.stream_video_noise_cancellation.NoiseCancellationAudioProcessingController
+
+/** 
+ * StreamVideoNoiseCancellationPlugin
+ * 
+ * A Flutter plugin that provides noise cancellation functionality for Stream Video.
+ */
+class StreamVideoNoiseCancellationPlugin: FlutterPlugin, MethodCallHandler {
+  private lateinit var channel: MethodChannel
+  private lateinit var applicationContext: Context
+  private var controller: NoiseCancellationAudioProcessingController? = null
+  private val TAG = "NoiseCancellationPlugin"
+  
+  override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
+    Log.d(TAG, "Plugin attached to engine")
+    channel = MethodChannel(flutterPluginBinding.binaryMessenger, "stream_video_noise_cancellation")
+    applicationContext = flutterPluginBinding.applicationContext
+    channel.setMethodCallHandler(this)
+  }
+
+  override fun onMethodCall(call: MethodCall, result: Result) {
+    Log.d(TAG, "Method call: ${call.method}")
+    when (call.method) {
+      "registerProcessor" -> {
+        try {
+          controller = NoiseCancellationAudioProcessingController(applicationContext)
+          FlutterWebRTCPlugin.sharedSingleton.setAudioProcessingFactoryProvider(controller)
+          Log.d(TAG, "Noise cancellation processor registered successfully")
+          result.success(null)
```

Comment:
Do we need a check if the controller is already set? Don't know, maybe dispose an existing controller or throw an error that you registered twice?